pallets/settlement/src/lib.rs

```diff
@@ -876,9 +881,9 @@ decl_module! {
         pub fn reschedule_instruction(origin, id: InstructionId) {
             let did = Identity::<T>::ensure_perms(origin)?;
 
-            <InstructionDetails<T>>::try_mutate(id, |details| {
-                ensure!(details.status == InstructionStatus::Failed, Error::<T>::InstructionNotFailed);
-                details.status = InstructionStatus::Pending;
+            <InstructionStatuses<T>>::try_mutate(id, |status| {
+                ensure!(*status == InstructionStatus::Failed, Error::<T>::InstructionNotFailed);
+                *status = InstructionStatus::Pending;
```

Comment:
Using `try_mutate` here is ok, since we want to check what the current status is before changing it.